webapp/app.py

```
@@ -779,5 +926,208 @@ def language(lang_code):
+    if os.path.exists(cache_path):
+        return app.send_static_file(f"word-images/{lang_code}/{word.lower()}.webp")
+
+    # HEAD requests just check cache — don't trigger generation
+    if request.method == "HEAD":
+        return "", 404
+
+    # Fetch definition to include in prompt (best-effort)
+    definition_hint = ""
+    try:
+        def_url = f"https://en.wiktionary.org/api/rest_v1/page/definition/{urllib.parse.quote(word.lower())}"
+        req = urlreq.Request(def_url, headers={"User-Agent": "WordleGlobal/1.0"})
```

Ruff (0.15.1)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)